lib/mongo/grid/stream/read.rb

```diff
@@ -167,7 +167,8 @@ def ensure_readable!
           end
 
           def view
-            @view ||= fs.chunks_collection.find({ :files_id => file_id }, options).read(read_preference).sort(:n => 1)
+            @view ||= (opts = options.merge(read: read_preference) if read_preference
```

Comment:
I wanted that whole block of code only executed once so I put it all in a memoized ()
``options.merge(read: read_preference)`` creates a new object and since the read object is immutable, we only have to do it once when the ``@view`` is created. Do you have another suggestion for how we can format this method to be more readable and so that the lines are only executed once?